src/views/clusteroverview/OverviewTab/metric-charts-card/utils/hooks/useMetricChartData.ts

```diff
@@ -0,0 +1,74 @@
+import { useMemo } from 'react';
+
+import { MILLISECONDS_MULTIPLIER } from '@kubevirt-utils/components/Charts/utils/utils';
+import DurationOption from '@kubevirt-utils/components/DurationOption/DurationOption';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { PrometheusEndpoint, usePrometheusPoll } from '@openshift-console/dynamic-plugin-sdk';
+import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk-internal';
+
+import { METRICS } from '../constants';
+import { getMetricQuery } from '../metricQueries';
+
+import { ChartData, ChartDomain } from './types';
+import { findUnit, getHumanizedValue, getLargestValue, getPointDay } from './utils';
+
+export type MetricChartData = {
+  chartData: ChartData;
+  domain: ChartDomain;
+  isReady: boolean;
+  largestValue: number;
+  numberOfTicks: number;
+  unit: string;
+};
+
+type UseMetricChartData = (metric: string) => MetricChartData;
+
+const useMetricChartData: UseMetricChartData = (metric) => {
+  const [activeNamespace] = useActiveNamespace();
+  const currentTime = useMemo(() => Date.now(), []);
+  const timespan = DurationOption.getMilliseconds(DurationOption.ONE_WEEK.toString());
+
+  const [queryData] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint.QUERY_RANGE,
+    query: getMetricQuery(metric, activeNamespace),
+    namespace: activeNamespace,
+    endTime: currentTime,
+    timespan,
+  });
+
+  const rawData = queryData?.data?.result?.[0]?.values;
+  let largestValue = getLargestValue(rawData);
```

Comment:
why the use of let?
const getLargestValuePhilip = (metric, rawData, unit) => {
const value = getLargestValue(rawData);
if (metric === METRICS.MEMORY || metric === METRICS.STORAGE) {
    return getHumanizedValue(metric, value, unit);
  }
return value;
}


const largestValue = getLargestValuePhilip(rawData);

Reply:
I worked it out a different way, but got rid of the `let`.